src/track/track.cpp

```diff
@@ -746,6 +746,42 @@ void Track::removeCue(const CuePointer& pCue) {
     emit(cuesUpdated());
 }
 
+void Track::removeMainCue() {
+    QMutexLocker lock(&m_qMutex);
+    for (const CuePointer& pCue : m_cuePoints) {
```

Comment:
Use QMutableListIterator and remove() through the iterator

Reply:
I tried that and it is not the issue. I think Track::removeHotCues is working because of some non-obvious side effect it has.